### PR TITLE
Enhanced object and database caches

### PR DESCRIPTION
**Pull request by jacobd91**
Updated to only clear by groups based on actions, to support CPTs, and to resolve issues surrounding certain hooks failing to trigger for CPTs when using the classic editor format

### CODE REVIEW

**Comment by maxicus** [comment]
Those hardcode of groups purging now will cause questions for sure. Also there are areas of conflict with objectcache.purge.all (which is not in UI whats bad) + there are cases when one just doesnt want to purge OC at all on posts changes relying on natural expiration when it works good enough.

I'm thinking what if add it in more advanced mode, and bound to events?
like 
Groups to flush 
- on posts change
- on comments activity
- on remaining events (like options change)

with possible value "<all>" in each list...

**Review by maxicus — CHANGES_REQUESTED** [review]
I see some extra complications = slowdowns added to dbcache, object cache, and removal of critical hooks like clean_post_cache (one called by wp to clean cache).
Effect is not straightforward. Can't assess if it a gain in average website env or slow down.

At least it's not solving my previous comment, but seems something else (not sure what). 
So... think we need to agree first what this PR is supposed to do enhance and in which cases.

**Review by cssjoe — CHANGES_REQUESTED** [review]
Max requested changes 12 days ago.  There are a lot of PHPCS WPCS violations.  Max mentioned if this update will increase performance.  We could profile the execution with XHPROF or similar.

File: ObjectCache_Plugin.php [hunk 1]
```diff
@@ -185,8 +185,16 @@ function on_post_change( $post_id = 0, $post = null ) {
 			}
 
 			$flush = Dispatcher::component( 'CacheFlush' );
-			$flush->objectcache_flush();
-			$flushed = true;
+			if( $this->_config->get_boolean( 'objectcache.purge.posts' ) ) {
```
[the remark this answers is not in the document]
Comment (jacobd91) [reply ObjectCache_Plugin.php]: Could you clarify on how this should be styled?

**Comment by jacobd91** [comment]
Abandoned

**Pull request closed without merging**